### PR TITLE
Add dagger example

### PR DESCRIPTION
**Pull request by jeromevdl**
*Issue #, if available:* #8

*Description of changes:* 
 - Add dagger example (standard JVM only)
 - Updated the load test results

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by alvarosanchez** [comment]
Dagger is no near comparable to any Java framework. Presenting as if it were apples to apples is quite unfair, IMHO

**Comment by jeromevdl** [comment]
> Dagger is no near comparable to any Java framework. Presenting as if it were apples to apples is quite unfair, IMHO

I agree Dagger does not provide all the features the others provide and is mainly a DI framework. 

But looking at the other examples here, this is the main feature we showcase (and maybe the specific Handler provided by some of them). DI is also probably the feature people are looking for first. And for this, Dagger is really good, really lightweight and not based on reflection, which makes it really performant. 

Other thing, there is no or very few examples of Lambda using Dagger, although everyone is looking for a way to reduce the cold start. Android developers massively rely on Dagger for his speed, and I think it makes sense for Lambda too. Even if not fully comparable, I think it makes sense to have a sample of it. Maybe we can add a disclaimer or something to avoid any confusion / misunderstanding.

**Comment by alvarosanchez** [comment]
I find reasonable to add a Dagger example. But perhaps its performance results should be displayed in a separate table

**Comment by muraria** [comment]
@jeromevdl Thanks for making this PR.
May be you can put an asterisk next to Dagger and add a note/disclaimer below the table calling out the difference in frameworks.
Also, other framework samples here has GraalVM versions of those applications. Can you also include that please?

**Comment by muraria** [comment]
Other samples has GraalVM equivalent applications. Can you add that as well please?

**Comment by jeromevdl** [comment]
> Other samples has GraalVM equivalent applications. Can you add that as well please?

I tried but it's not as simple as the others... I'll have to come back to this later.

File: dagger/load-test/run-load-test-native.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+STACK_NAME=dagger-example-native
```
Comment (dmahapatro) [inline dagger/load-test/run-load-test-native.sh:1]: If GraalVM native option is not ready yet, do we need this script at the moment?

File: dagger/pom.xml [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+          <groupId>software.amazon.awssdk</groupId>
+          <artifactId>apache-client</artifactId>
+        </exclusion>
+        <exclusion>
+          <groupId>software.amazon.awssdk</groupId>
+          <artifactId>netty-nio-client</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
```
Comment (dmahapatro) [inline dagger/pom.xml:83]: AWS CRT client? Did it not perform better compared to url connection client?

File: dagger/load-test/run-load-test-native.sh (hunk 1 above)
Comment (jeromevdl) [reply dagger/load-test/run-load-test-native.sh:1]: no, will remove

File: dagger/pom.xml (hunk 2 above)
Comment (jeromevdl) [reply dagger/pom.xml:83]: CRT client is still in preview. I agree once it becomes GA, we can move to it.

**Review by muraria — APPROVED** [review]
Approving. We will have to remove the Dagger load test folder in subsequent PR.

**Pull request merged**